src/neo/Cryptography/Crypto.cs

```diff
@@ -59,21 +62,50 @@ public static byte[] Sign(byte[] message, byte[] prikey, byte[] pubkey)
         /// <returns><see langword="true"/> if the signature is valid; otherwise, <see langword="false"/>.</returns>
         public static bool VerifySignature(ReadOnlySpan<byte> message, ReadOnlySpan<byte> signature, ECC.ECPoint pubkey)
         {
-            ECCurve curve =
-                pubkey.Curve == ECC.ECCurve.Secp256r1 ? ECCurve.NamedCurves.nistP256 :
-                pubkey.Curve == ECC.ECCurve.Secp256k1 ? ECCurve.CreateFromFriendlyName("secP256k1") :
-                throw new NotSupportedException();
-            byte[] buffer = pubkey.EncodePoint(false);
-            using var ecdsa = ECDsa.Create(new ECParameters
+            if (signature.Length != 64) return false;
+
+            if (IsOSX && pubkey.Curve != ECC.ECCurve.Secp256r1)
             {
-                Curve = curve,
-                Q = new ECPoint
+                try
                 {
-                    X = buffer[1..33],
-                    Y = buffer[33..]
+                    var curve = Org.BouncyCastle.Asn1.Sec.SecNamedCurves.GetByName(pubkey.Curve == ECC.ECCurve.Secp256r1 ? "secp256r1" : "secp256k1");
+                    var domain = new Org.BouncyCastle.Crypto.Parameters.ECDomainParameters(curve.Curve, curve.G, curve.N, curve.H);
+                    var point = curve.Curve.CreatePoint(
+                        new Org.BouncyCastle.Math.BigInteger(pubkey.X.Value.ToString()),
+                        new Org.BouncyCastle.Math.BigInteger(pubkey.Y.Value.ToString()));
+                    var pubKey = new Org.BouncyCastle.Crypto.Parameters.ECPublicKeyParameters("ECDSA", point, domain);
+                    var signer = Org.BouncyCastle.Security.SignerUtilities.GetSigner("SHA-256withECDSA");
+
+                    signer.Init(false, pubKey);
+                    signer.BlockUpdate(message.ToArray(), 0, message.Length);
+
+                    var sig = signature.ToArray();
+                    var r = new Org.BouncyCastle.Math.BigInteger(sig.AsSpan().Slice(0, 32).ToArray());
```

Comment:
```suggestion
                    var r = new Org.BouncyCastle.Math.BigInteger(1, sig, 0, 32);
                    var s = new Org.BouncyCastle.Math.BigInteger(1, sig, 32, 32);
```

Reply:
If we just convert bytes into Biginteger, `r` or `s` could be negative which is illegal for ECDSA, so we have to make it positive by passing 1 to the first parameter.